### PR TITLE
feat: added query pool 

### PR DESCRIPTION
**Pull request by gyuguen**
Added a function to query the pool.

### CODE REVIEW

**Review by inchori — APPROVED** [review]
LGTM 👍

How about we add query pool test in `grpc_query_pool_test.go` too?

File: x/datapool/types/errors.go [hunk 1]
```diff
@@ -11,4 +11,5 @@ var (
 	ErrNotEnoughPoolDeposit       = sdkerrors.Register(ModuleName, 3, "the balance is not enough to make a data pool")
 	ErrNotRegisteredDataValidator = sdkerrors.Register(ModuleName, 4, "data validator is not registered")
 	ErrNoRegisteredNFTContract    = sdkerrors.Register(ModuleName, 5, "no NFT contract is registered")
+	ErrNotCreatedPool             = sdkerrors.Register(ModuleName, 6, "pool not created")
```
Comment (hansol-medi) [inline x/datapool/types/errors.go:14]: Why don't we rename this to `ErrPoolNotFound`?

**Review by hansol-medi — APPROVED** [review]
LGTM

**Comment by gyuguen** [comment]
> LGTM 👍
> 
> How about we add query pool test in `grpc_query_pool_test.go` too?
Great. I'll add it

File: x/datapool/types/errors.go (hunk 1 above)
Comment (gyuguen) [reply x/datapool/types/errors.go:14]: Good. I'll fixed it

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Pull request merged**